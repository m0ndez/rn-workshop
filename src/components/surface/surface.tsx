import { Component, ComponentProps, ElementRef, forwardRef } from "react";
import { StyleSheet } from "react-native";
import { Surface as PaperSurface } from "react-native-paper";

type SurfaceProps = ComponentProps<typeof PaperSurface>;

type SurfaceRef = ElementRef<typeof PaperSurface>;

export const Surface = forwardRef<SurfaceRef, SurfaceProps>((props, ref) => {
  const { style, ...otherProps } = props;
  return (
    <PaperSurface ref={ref} style={[styles.root, style]} {...otherProps} />
  );
});

const styles = StyleSheet.create({
  root: {
    padding: 16,
    borderRadius: 8,
  },
});
