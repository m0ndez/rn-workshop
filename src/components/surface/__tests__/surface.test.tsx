import React from "react";
import { render } from "@testing-library/react-native";
import { Surface } from "../surface";
import { Text } from "react-native";

describe("Surface component", () => {
  it("should render correctly", () => {
    const { toJSON } = render(
      <Surface children={<Text>Snap Shot Test</Text>} />
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
