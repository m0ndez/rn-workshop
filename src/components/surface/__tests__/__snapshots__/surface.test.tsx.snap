// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Surface component should render correctly 1`] = `
<View
  collapsable={false}
  style={
    {
      "backgroundColor": "rgb(247, 243, 249)",
      "borderRadius": 8,
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.15,
      "shadowRadius": 3,
    }
  }
  testID="surface-outer-layer"
>
  <View
    collapsable={false}
    style={
      {
        "backgroundColor": "rgb(247, 243, 249)",
        "borderRadius": 8,
        "flex": undefined,
        "padding": 16,
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 0.3,
        "shadowRadius": 1,
      }
    }
    testID="surface"
  >
    <Text>
      Snap Shot Test
    </Text>
  </View>
</View>
`;
